### PR TITLE
Bringing some readability to prices and flows

### PR DESCRIPTION
**Pull request by xavierlepretre**
@rogersanick What do you think?

### CODE REVIEW

File: contracts/src/main/kotlin/com/contractsAndStates/contracts/BuildPhaseContract.kt [hunk 1]
```diff
@@ -28,273 +27,276 @@ class BuildPhaseContract : Contract {
...
-                val relevantHexTileNeighbours: ArrayList<HexTile?> = arrayListOf()
-
-                // Add neighbouring hexTiles to the storage array if they exist.
-                inputGameBoardState.hexTiles.get(hexTileIndex).sides.getNeighborsOn(hexTileCoordinate.getAdjacentSides())
-                        .forEach {
-                            if (it != null) relevantHexTileNeighbours.add(inputGameBoardState.hexTiles.get(it))
+                val itAndNeighboringCorners = inputBoard.hexTiles
+                        .getItAndNeighboringCorners(onCorner)
+                "A settlement must not have previously been built in this vicinity." using
+                        itAndNeighboringCorners.none { inputBoard.settlementsPlaced.hasOn(it) }
```
Comment (xavierlepretre) [inline contracts/src/main/kotlin/com/contractsAndStates/contracts/BuildPhaseContract.kt]: @rogersanick Example of improved readability of the whole lot. Not finished.
@abellagonzalo Example of usage of the traversing functions.

**Comment by xavierlepretre** [comment]
@rogersanick A lot of work to get the flows more straightforward.
Let's hope it does not cause too many conflicts with others...

**Review by rogersanick — COMMENTED** [review]
@xavierlepretre - let me know when this is ready to merge! LGTM

Assuming unit tests (mostly) pass?

**Comment by xavierlepretre** [comment]
The unit tests of states all pass. As for the flows, you have to change to `MutableList`, which I can add, for them to not be stuck on deserialisation.

**Comment by xavierlepretre** [comment]
I think you should merge Gonzalo's first. I will fix the conflicts on mine.

**Comment by rogersanick** [comment]
Merged @abellagonzalo s branch. Once you've fixed the conflicts I'll merge this PR as well. See you both tomorrow at CCC!

**Comment by xavierlepretre** [comment]
That was fast!

**Pull request merged**